### PR TITLE
[BB-6151] Update TinyMCE Editor static file references to upgraded version 5.x

### PR DESCRIPTION
**Pull request by tecoholic**
**TL;DR -** Replaces the version 4.x references of TinyMCE static files with version 5.x and removes the workaround required for TMCE 4 from oa_editor_tinymce.js

JIRA: - NA

**What changed?**

The edx-platform's TinyMCE is being upgraded from version 4.x to 5.x (v5.5.1) at https://github.com/openedx/edx-platform/pull/30335. The upgrade breaks the ORA WYSIWYG editor as the TinyMCE files referenced in the [oa_editor_tinymce.js](https://github.com/openedx/edx-ora2/blob/master/openassessment/xblock/static/js/src/lms/editors/oa_editor_tinymce.js#L7) are hard-coded to v4.x file paths.

- In this PR these paths are update to reference the v5.x files
- Corresponding changes are made in the TinyMCE init function
- Removes a TinyMCE 4.x specific workaround
- Updates all the static files using `make static`

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [ ] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**

1. Checkout to the PR branch of https://github.com/openedx/edx-platform/pull/30335
2. Add a ORA block in a course and set the editor to WYSIWYG
3. Open the Unit in the LMS and notice that the Editor doesn't load for input
4. Install the `open-craft:tecoholic/BB-6151-upgrade-tinymce-to-v5` branch of the ORA in to lms
5. Restart the LMS
6. Revisit the test unit in LMS and verify that the Editor is functioning as expected

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @tecoholic!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by Agrendalath** [comment]
@natabene, would you mind approving the CI on this PR?

**Review by Agrendalath — COMMENTED** [review]
@tecoholic, while the LMS is working for me, I cannot use it with Studio on my devstack. When I switch the "Response Editor" to "WYSIWYG Editor", I'm getting the following warning:
```js
Source map error: Error: request failed with status 404
Resource URL: http://localhost:18000/static/dist/openassessment-lms.d67e4d76d91bec14c4f5.js
Source Map URL: openassessment-lms.d67e4d76d91bec14c4f5.js.map
```

Then, I'm getting a 404 on http://localhost:18010/static/studio/bundles/js/factories/themes/silver/theme.js. The editor doesn't load at all there.

I'm running `make dev.static.studio` after installing this version of XBlock. Restarting the Studio container doesn't help too. Am I doing something incorrectly?

Also, please bump the version in `__init__.py` and in `package.json`.

**Comment by tecoholic** [comment]
@Agrendalath The source file missing error seems to because of them being [gitignored](https://github.com/openedx/edx-ora2/commit/6e0a4dc372c602d70e3d20d509b9ae426ab150b0). So you will have to do a npm build locally if you want then to load I guess. However, the theme file missing is an issue. Let me figure that out and let you know.

**Comment by tecoholic** [comment]
@Agrendalath The issue is now fixed with [2c3dea8](https://github.com/openedx/edx-ora2/pull/1869/commits/2c3dea8bdf11206d3583ba3b4a4a674cb89cf1c2).

**Comment by tecoholic** [comment]
@Agrendalath I see. I am guessing the theme is not getting properly set for the `openassessment-studio.js` file. I will find and fix and get back to you.

**Comment by tecoholic** [comment]
I found another file initializing the TinyMCE with its own configuration. Now the editor should work in:
* LMS
* Studio preview of the Block
* Edit window of the Studio

**Comment by Agrendalath** [comment]
@tecoholic,
1. I'm now getting 404 for `https://studio.bb-6151-sandbox.opencraft.hosting/static/studio/bundles/js/factories/themes/silver/theme.js`. This seems to be breaking the JS when the `Simple Text Editor` is enabled - the simple editor is not displayed for prompts. Please see the editor of the second XBlock on [this page](https://studio.bb-6151-sandbox.opencraft.hosting/container/block-v1:edX+DemoX+Demo_Course+type@vertical+block@fb79dcbad35b466a8c6364f8ffee9050).
1. Nit: we should remove the `textcolor` plugin from `openassessment/xblock/static/js/src/studio/oa_tiny_mce.js`, as it's already built into the core editor.

**Review by Agrendalath — APPROVED** [review]
@tecoholic, please bump the version in `__init__.py` and `package.json`.

:+1:

* [x] I tested this: tested the XBlock on [the sandbox](https://bb-6151-sandbox.opencraft.hosting) and locally
* [x] I read through the code
* [x] I checked for accessibility issues: n/a
* [x] Includes documentation: n/a
* [x] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository: n/a

**Comment by Agrendalath** [comment]
@natabene, this is ready for your review. It's a dependency for https://github.com/openedx/edx-platform/pull/30335.

**Comment by tecoholic** [comment]
Version bumped to 4.5.0

**Comment by natabene** [comment]
@tecoholic Thank you for your contribution. @Agrendalath Got it, thanks.

**Review by pomegranited — APPROVED** [review]
:+1: Works great @tecoholic ! And thank you for providing such complete testing instructions.

I noticed one issue which would warrant a follow-up ticket: Using the WYSIWYG editor + Allowing LaTeX responses doesn't work well -- there's no way to get a good LaTeX preview with the TinyMCE editor.

WYSIWYG+LaTeX is bad

- [x] I tested this on my devstack using the PR testing instructions.
- [x] I read through the code.
- [x] I checked for accessibility issues by following the [TinyMCE accessibility instructions](https://www.tiny.cloud/docs/advanced/accessibility/).
- [x] <strike>Includes documentation</strike> N/A
- [x] Commit structure follows [OEP-0051](https://github.com/edx/open-edx-proposals/pull/189)

**Comment by pomegranited** [comment]
@tecoholic Could you resolve the conflicts here to prepare this for merge?

Once that's done, I can merge and tag this if that's ok with you @Agrendalath ?

**Comment by natabene** [comment]
@pomegranited Before you merge, can you please give heads up or coordinate in  cc-core-applications Slack channel?

**Comment by tecoholic** [comment]
@tecoholic Sure. I will resolve the conflicts and let you know. :+1:

**Comment by tecoholic** [comment]
@pomegranited I have rebased the changes on to the master and resolved the conflicts. :)

**Comment by pomegranited** [comment]
Thanks @tecoholic :)

Hi @openedx/content-aurora -- does anyone have any objections to merging this change?
If not, I'll merge & tag it after [midnight 9 Aug UTC](https://www.timeanddate.com/worldclock/fixedtime.html?msg=merge+ORA2%231869&iso=20220809T0930&p1=5).

**Comment by tecoholic** [comment]
@pomegranited @Agrendalath The CI tests are failing when installing JS dependencies as the [@edx/paragon](https://github.com/openedx/paragon) depends on React 16.x while `edx-ora2` depends on React 17.x. 

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @edx/paragon@12.8.0
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"^17.0.2" from the root project
npm ERR!   peer react@">=16.x" from @fortawesome/react-fontawesome@0.1.14
npm ERR!   node_modules/@fortawesome/react-fontawesome
npm ERR!     @fortawesome/react-fontawesome@"^0.1.11" from @edx/paragon@12.8.0
npm ERR!     node_modules/@edx/paragon
npm ERR!       @edx/paragon@"^12.8.0" from the root project
npm ERR!   18 more (@restart/context, @restart/hooks, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.6" from @edx/paragon@12.8.0
npm ERR! node_modules/@edx/paragon
npm ERR!   @edx/paragon@"^12.8.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: react@16.14.0
npm ERR! node_modules/react
npm ERR!   peer react@"^16.8.6" from @edx/paragon@12.8.0
npm ERR!   node_modules/@edx/paragon
npm ERR!     @edx/paragon@"^12.8.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

The current version of `@edx/paragon` is 20.8.0. Can we update the version of `paragon` in ORA2 to a more recent version with React 17 support? I am not aware of the impact of upgrading Paragon. Kindly advise.

**Comment by pomegranited** [comment]
Sorry @tecoholic, I wasn't aware of this build issue, but I do think that upgrading paragon to the latest possible version is the way to go here.

You'll have to go through the [paragon release notes](https://github.com/openedx/paragon/releases) to find out the impact of upgrading from 12.8.0 to 20.x.x. E.g., [v20.0.0 has breaking changes](https://github.com/openedx/paragon/releases/tag/v20.0.0), which look involved but well-documented.

**Comment by tecoholic** [comment]
The upgrade of Paragon to 20.x is blocked by https://github.com/openedx/paragon/issues/1559

cc: @Agrendalath @pomegranited

**Comment by tecoholic** [comment]
@Agrendalath @pomegranited I have updated the [openedx/paragon](openedx/paragon) to the latest version `20.9.2` with the bug fix for previously identified SCSS issue. Can one of you approve the workflows for the CI?

**Comment by Agrendalath** [comment]
@tecoholic, I'll leave triggering the CI to @pomegranited or @natabene, as I don't have access to this repo.

**Comment by tecoholic** [comment]
@pomegranited @natabene I have added the fix for the failing testing. Kindly approve the workflow when you get some time.

**Comment by tecoholic** [comment]
@pomegranited The translations seems to be failing due the changes the filename of the newly compiled JS files. I tried to update the translation files by pulling the latest from Transifex, but ran into the following error. And I am also not sure what the right procedure is to do this. Neither the [ORA dev docs](https://openedx.atlassian.net/wiki/spaces/AC/pages/9765004/ORA+Developer+Guide) or the [Translation Management](https://openedx.atlassian.net/wiki/spaces/AC/pages/1984790989/Translation+Management) were helpful in this aspect. Can you kindly help here?

```
$ make pull_translations
cd ./openassessment/ && tx pull -a -f --mode reviewed --minimum-perc=1
# Getting info about resources

edx-platform.openassessment - Get "/resources/o:open-edx:p:edx-platform:r:openassessment": unsupported protocol scheme ""
edx-platform.openassessment-js - Get "/resources/o:open-edx:p:edx-platform:r:openassessment-js": unsupported protocol scheme ""
[##############################] (2 / 2)
Aborted
make: *** [Makefile:99: pull_translations] Error 1
```

**Comment by pomegranited** [comment]
@tecoholic Hmm.. Those `o:open-edx:p:edx-platform:r:openassessment` lines are coming from the [`.tx/config`](https://github.com/openedx/edx-ora2/blob/master/.tx/config) files, and [transifex's docs for tx/config](https://docs.transifex.com/client/client-configuration#-tx/config) say they expect a `project_slug.resource_slug`, not a thing with colons in it. But that format matches the [edx-platform `.tx/config`](https://github.com/openedx/edx-platform/blob/master/.tx/config), and is a relatively recent change here: 

https://github.com/openedx/edx-ora2/pull/1826

@UsamaSadiq as author of ^ can you help with this issue?

**Comment by UsamaSadiq** [comment]
Hi @pomegranited 
Seems like the error is occurring due to the new transifex client which was upgraded recently. I'll have a detailed look at the issue and will reach back to you soon enough.

**Comment by pomegranited** [comment]
Hi @UsamaSadiq , do you have any updates here? Is there anything I can help with?

**Comment by UsamaSadiq** [comment]
Sorry, I got busy in some priority tasks and couldn't get back to this. I'll investigate this issue and will update you by tomorrow for sure.

**Comment by UsamaSadiq** [comment]
Added new command in `Makefile` to install the latest transifex client locally for running translation commands in the PR https://github.com/openedx/edx-ora2/pull/1889.

**Comment by UsamaSadiq** [comment]
HI @pomegranited, 
I've created final PR https://github.com/openedx/edx-ora2/pull/1890 to make the `tx` command available on local with the new `transifex` client. 
Since your PR is from a fork, so I couldn't check it out on local to debug. After investigating, I'd suggest you to perform following steps:
- Rebase your PR to get the new make target in your branch.
- Run the `make install_transifex_client` command which will download the latest `GO Transifex` client and make it accessible across the repo. 
- Run `make pull_translations` command to fetch the latest translations from `Transifex` in case any of the translations need to be updated. 
- Run `tox -e quality` to verify if the check passes now with the updated translations.

I'm hoping that this'll resolve the issue with the translations failure but I'll be able to help further after knowing the result of these instructions.

**Comment by pomegranited** [comment]
Thank you @UsamaSadiq !
@tecoholic  could you update this PR as described above?

**Comment by tecoholic** [comment]
@pomegranited I was getting the `unsupported protocol scheme` errors after following the steps as well. Turns out, my Transifex client configuration was incompatible with the Go Client and I needed to [migrate my old transifex client config](https://docs.transifex.com/cli-client-in-go/introduction-to-the-client#migrating-from-older-versions-of-the-client) using the `tx migrate` command before `make pull_translations` could start working.

However I have run into another issue. All the files that are being pulled from transifex just has the following:

```
{"errors":[{"status":401,"code":"unauthorized","title":"Unauthorized","detail":"Invalid JWT token"}]
```
I was using the OC account and even created a new API token to double check it is not due to some invalid token.

@UsamaSadiq Do you have any idea why this is happening?

Also I had to change the Makefile command for `pull_translations` as it was creating a different directory `openassessment/conf/` and kept downloading the files into it.

```diff
pull_translations: ## pull translations from Transifex
- 	cd ./openassessment/ && tx pull -a -f --mode reviewed --minimum-perc=1
+	tx pull -a -f --mode reviewed --minimum-perc=1
```

**Comment by Agrendalath** [comment]
@tecoholic, this is working correctly for me on your branch. Please make sure that your `~/.transifexrc` looks like this:
```ini
[https://www.transifex.com]
api_hostname = https://api.transifex.com
hostname = https://www.transifex.com
password = <TOKEN>
rest_hostname = https://rest.api.transifex.com
token = <TOKEN>
username = API
```

I've shared the token with you through a different channel.

**Comment by tecoholic** [comment]
@Agrendalath Thank you. I also have the same version of Transifex. And everything looks the same in my transifex file except the username which is in lowercase `api` instead of the uppercase `API` as you have mentioned. I think this might be issue. Let me give it another try.

**Comment by tecoholic** [comment]
@Agrendalath I tried with the uppercase username and the API key you shared. I get the same results as before. Are you able to run `make pull_translations` and `make compile_translations` without errors? I get a number of these 

```
Execution of msgfmt failed: /src/edx-ora2/openassessment/conf/locale/el/LC_MESSAGES/django.po:1:2: syntax error
msgfmt: found 1 fatal error
Execution of msgfmt failed: /src/edx-ora2/openassessment/conf/locale/fr_CA/LC_MESSAGES/djangojs.po:1:2: syntax error
msgfmt: found 1 fatal error
Execution of msgfmt failed: /src/edx-ora2/openassessment/conf/locale/lv/LC_MESSAGES/django.po:1:2: syntax error
msgfmt: found 1 fatal error
Execution of msgfmt failed: /src/edx-ora2/openassessment/conf/locale/lv/LC_MESSAGES/djangojs.po:1:2: syntax error
msgfmt: found 1 fatal error
Execution of msgfmt failed: /src/edx-ora2/openassessment/conf/locale/fr/LC_MESSAGES/django.po:1:2: syntax error
msgfmt: found 1 fatal error
Execution of msgfmt failed: /src/edx-ora2/openassessment/conf/locale/fr/LC_MESSAGES/djangojs.po:1:2: syntax error
msgfmt: found 1 fatal error
```

And when I open one of the mentioned files in the errors, I see this

```
{"errors":[{"status":401,"code":"unauthorized","title":"Unauthorized","detail":"Invalid JWT token"}]}
```

**Comment by Agrendalath** [comment]
@tecoholic, huh, that's odd. I tried this again and could reproduce it for some translations without an underscore (`_`) in them. However, after I tested it with `tx version 1.0.3` and switched back to the newer one, it worked fine. I've allowed myself to push them to your branch.

There were some errors in the `it_IT` translation:
```python
INFO:i18n.validate:

WARNING:i18n.validate:
it_IT/LC_MESSAGES/django.po:1411: 'msgid' and 'msgstr' entries do not both begin with '\n'
it_IT/LC_MESSAGES/django.po:1435: 'msgid' and 'msgstr' entries do not both begin with '\n'
it_IT/LC_MESSAGES/django.po:1499: 'msgid' and 'msgstr' entries do not both begin with '\n'
msgfmt: found 3 fatal errors
```
I've added translation suggestions [to Transifex](https://www.transifex.com/open-edx/edx-platform/translate/#it_IT/openassessment/331159809?q=check%3Aerror), but I don't have permissions to accept them (cc: @UsamaSadiq). As a workaround, I've manually fixed these strings in the translation files.

@UsamaSadiq, many languages appear to be duplicated in Transifex. E.g. in [the resources](https://www.transifex.com/open-edx/edx-platform/openassessment/) we can see `pl` and `pl_PL`, which are the same language. This applies to most languages. When I tried using an older Transifex client, all of them were being downloaded. Is it something we could fix to avoid confusion?

**Comment by Agrendalath** [comment]
Update: since the CI was failing, I've also rebuilt statics (just in case) and updated source translations. However, I don't have permission to update them in Transifex, as `make push_translations` is returning this:
```python
tx push -s
# Getting info about resources

edx-platform.openassessment - Error while fetching stats, 404, not_found: No `Project` resource found with the identifier `o:open-edx:p:edx-platform`
edx-platform.openassessment-js - Error while fetching stats, 404, not_found: No `Project` resource found with the identifier `o:open-edx:p:edx-platform`
[##############################] (2 / 2)
Aborted
make: *** [Makefile:102: push_translations] Error 1
```

cc: @UsamaSadiq

**Comment by tecoholic** [comment]
@Agrendalath Thanks you for finding the solution for this. Hopefully we will be able to wrap up this PR with this.

**Comment by pomegranited** [comment]
@tecoholic @Agrendalath I don't have push rights on the Transifex projects either, but it looks like we have a bot to take care of that: https://github.com/openedx/edx-ora2/pulls?q=is%3Apr+author%3Aedx-transifex-bot+

@tecoholic  There's a couple conflicts remaining, but once they're resolved, I can merge this. Can we aim for first thing UTC Thu 15 Sep?

**Comment by tecoholic** [comment]
@pomegranited I have rebased the PR branch on master, resolved conflicts, rebuilt the static files and translations and pushed the changes. I have also run `tox -e quality` to verify there aren't any quality issues. Kindly approve the workflows and we should be able to merge this.

**Comment by openedx-webhooks** [comment]
@tecoholic 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-ora2&entry.867055334=https://github.com/openedx/edx-ora2/pull/1869&entry.1484655318=https://github.com/tecoholic&entry.752974735=2022-07-01+06:32&entry.1917517419=2022-09-14+02:02&entry.2133058324=Yes) so we can improve your experience in the future.
